### PR TITLE
[fix](stats) Fix thread leaks when doing checkpoint

### PR DESCRIPTION
**Pull request by Kikyou1997**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisTaskExecutor.java [hunk 1]
```diff
@@ -50,19 +50,12 @@ public AnalysisTaskExecutor(int simultaneouslyRunningTaskNum) {
                     TimeUnit.DAYS, new LinkedBlockingQueue<>(),
                     new BlockedPolicy("Analysis Job Executor", Integer.MAX_VALUE),
                     "Analysis Job Executor", true);
+            cancelExpiredTask();
```
Comment (Jibing-Li) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisTaskExecutor.java:53]: This cancelExpiredTask will only executed once?

Comment (Kikyou1997) [reply fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisTaskExecutor.java:53]: It would submit a thread to do this job

Comment (Jibing-Li) [reply fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisTaskExecutor.java:53]: I see, cancelExpiredTask this function will create a thread.
I think we need to sleep for a while between two iterations, or this thread may use too much cpu resource.

Comment (Kikyou1997) [reply fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisTaskExecutor.java:53]: It try to take elements from a blocking queue and then block until timeout, so won't cost too much CPU

**Review by Jibing-Li — APPROVED** [review]
LGTM

**Comment by Kikyou1997** [comment]
run buildall

**Review by englefly — DISMISSED** [review]

**Comment by Kikyou1997** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Pull request merged**